src/stability.md

```diff
@@ -0,0 +1,35 @@
+# Stability
+
+This section is about the stability attributes and schemes that allow stable APIs to use unstable
+APIs internally in the rustc standard library.
+
+For instructions on stabilizing a feature see [Stabilizing Features](./stabilization_guide.md).
+
+# unstable
+
+The `#[unstable(feature = "foo", issue = "1234", reason = "lorem ipsum")]` attribute explicitly
+marks an item as unstable. This infects all sub-items, where the attribute doesn't have to be
```

Comment:
should I be explaining that? I assumed that to be obvious, but since you're asking it seems that it is not. It's essentially the same rules that `pub` has

Reply:
Probably a good idea to add a note re. "essentially `pub`".

Reply:
The first example makes some amount of sense to me intuitively, but the second doesn't. How do I use a stable function inside of an unstable module? Wouldn't I get a warning just by accessing the module to import the function?

Reply:
similar to `pub`, this pattern is often used when the function is reexported elsewhere, but the module is still unstable

Reply:
I think this is worth explaining more, actually. In particular, there is a limitation here in our stability check which *does* make the location of the stable item in the unstable module available to stable code, so the module cannot be renamed or removed any more. The document should note that problem.

Reply:
The relevant Rust bug is https://github.com/rust-lang/rust/issues/15702.